// @twolions/linkedin-private-api/dist/src/requests/profile.request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProfileRequest = void 0;
class ProfileRequest {
    constructor({ request }) {
        this.request = request;
    }
    getProfile({ publicIdentifier }) {
        const queryParams = {
            q: 'memberIdentity',
            memberIdentity: publicIdentifier,
            decorationId: 'com.linkedin.voyager.dash.deco.identity.profile.TopCardSupplementary-100',
        };
        return this.request.get('identity/dash/profiles', {
            params: queryParams,
        });
    }
    getOwnProfile() {
        return this.request.get('me');
    }
    getContactInfo({ publicIdentifier }) {
        return this.request.get(`identity/profiles/${publicIdentifier}/profileContactInfo`);
    }
    follow({ profileId }) {
        const requestPayload = {
            patch: { "$set": { following: true } }
        };
        this.request.post(`feed/dash/followingStates/urn:li:fsd_followingState:urn:li:fsd_profile:${profileId}`, requestPayload);
    }
    unfollow({ profileId }) {
        const requestPayload = {
            patch: { "$set": { following: false } }
        };
        this.request.post(`feed/dash/followingStates/urn:li:fsd_followingState:urn:li:fsd_profile:${profileId}`, requestPayload);
    }
}
exports.ProfileRequest = ProfileRequest;
